cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +740,73 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(metaPath, userPath):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              metaPath, userPath)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(metaPath))
+
+    if 'network' in md:
+        network = md['network']
+        del md['network']
+
+    if userPath:
+        ud = util.load_file(userPath).replace("\r", "")
+    return md, ud, cfg, network
+
+
+def load(data):
```

Comment:
might be able to drop in favor of util.read_conf which behaves with this default {} for unparseable or empty files.

Reply:
util.read_conf() doesn't raise exception if it's not a valid yaml, it will log a warning trace and return {}. But here we want to catch the format exception and ask our vSphere user to modify it.

Reply:
+1 let's rename this function to give a bit more clarity:
```suggestion
def safeload_yaml_or_dict(data):
```